client/replays/action-creators.ts

```diff
@@ -15,11 +15,8 @@ import { makeServerUrl } from '../network/server-url'
 
 const ipcRenderer = new TypedIpcRenderer()
 
-// TODO(tec27): Move this to an IPC call
-// TODO(tec27): Tighten up the types in here once the dependencies and actions have been migrated
-// to TS
 async function getReplayHeader(filePath: string): Promise<ReplayHeader | undefined> {
-  return ipcRenderer.invoke('replayParseHeader', filePath)
+  return (await ipcRenderer.invoke('replayParseData', filePath))?.headerData
```

Comment:
I don't think this action should exist in this form any more, it just encourages bad practices in reading replays (duplicating reads unnecessarily). I don't really see why this action needs to exist at all any more, if we're loading a replay from the replay browser it already has the header

Reply:
Hmm, not sure what exactly you mean by "action" here (redux action?), but the replay browser only has the header if the user expands the info panel. They can still watch the replay without parsing the header in the replay browser. Otherwise, we'd have to parse the header of every replay in the browser, right?

Reply:
This function is in `action-creators` so it is seemingly an action (or part of an action).

There's no need to deal with every replay, you only need the header from replays when
1) the Watch Replay button is clicked
2) the file thing is expanded

imo the former can be handled with an optional header data, and *only* do the IPC if we don't have it. If you don't want to do that, then at the very least please inline this function into the place that it is called, the goal is to not provide functions that look "convenient" but are potentially quite expensive